### ISSUE TITLE
Only DataSource and Cache keyworks accepted on SqliteConnectionStringBuilder

### ISSUE THREAD

**Issue by brunoklein99** [aspnet/Microsoft.Data.Sqlite#177]
Currently there are only two keywords available, `DataSource` and `Cache`. What about the [others](https://www.connectionstrings.com/sqlite-net-provider/)?

**Comment by bricelam** [comment]
This is not SQLite.NET (aka System.Data.SQLite). What features in particular would you like to see added to this library?

**Comment by brunoklein99** [comment]
The main one that I need are 'Password', 'Cache Size', 'Page Size', 'Journal Mode' and 'Synchronous'.

**Comment by ErikEJ** [comment]
You can issue PRAGMAs for most of those: https://www.sqlite.org/pragma.html

**Comment by brunoklein99** [comment]
Nice. Thank you for the input. Although for my next project security is above performance. Any idea on the database password/encryption ETA?

**Comment by ErikEJ** [comment]
Encryption is not built into SQLite, and you can load encryption extensions via PRAGMAs as described here: https://www.sqlite.org/see/doc/trunk/www/readme.wiki

**Comment by bricelam** [comment]
We may need to expose API for `sqlite3_enable_load_extension` before this will work. Until then, you can call it manually using code like the following.

```C#
static class SqliteConnectionExtensions
{
    [DllImport("sqlite3", CallingConvention = CallingConvention.Cdecl)]
    private static extern int sqlite3_enable_load_extension(IntPtr db, int onoff);

    public static void EnableExtensions(this SqliteConnection connection)
        => sqlite3_enable_load_extension(connection.Handle, true);
}
```

**Comment by brunoklein99** [comment]
It seems SQLite SEE costs $2000,00. Is it true?

**Comment by natemcmaster** [comment]
[So it would seem](http://www.hwaci.com/sw/sqlite/see.html). AFAIK we do not plan to ship SEE in this package. It looks like enabling this extension is more than just a matter of loading. We would take a PR to add calls necessary to make this work.

**Comment by brunoklein99** [comment]
Any viable option for SQLite encryption with EF7 in the near future? Or another database which requires no installation.

**Comment by natemcmaster** [comment]
Can you add a [new issue to Entity Framework](https://github.com/aspnet/EntityFramework/issues/new) for your request for encryption? (This request may be solved by allowing the use of System.Data.Sqlite instead of this repo.) 

Btw, @ErikEJ created a provider for SQL CE (a file-based DB similar to SQLite) [EntityFramework.SqlServerCompact40](http://docs.efproject.net/en/latest/providers/index.html#entityframework-sqlservercompact40). IIRC this supports encyption.

**Comment by ErikEJ** [comment]
Yes, the SQL CE provider supports encryption out of the box. And runs on Windows x86 and x64

**Comment by bricelam** [comment]
It looks like System.Data.SQLite's encryption will only work on Windows.

After a quick search, it looks like there are many alternatives to SSE.

**Comment by bricelam** [comment]
I've implemented the extension loading in #183
